src/storage/kv/mod.rs

```diff
@@ -646,10 +646,22 @@ impl<I: Iterator> Cursor<I> {
     pub fn valid(&self) -> Result<bool> {
         if !self.iter.valid() {
             if let Err(e) = self.iter.status() {
-                CRITICAL_ERROR.with_label_values(&["rocksdb"]).inc();
+                CRITICAL_ERROR.with_label_values(&["rocksdb iter"]).inc();
                 if panic_when_unexpected_key_or_data() {
                     set_panic_mark();
-                    panic!("Rocksdb error: {}", e);
+                    panic!(
```

Comment:
Why panic?

Reply:
the error is unexpected and by default `panic_when_unexpected_key_or_data` will return false